### PR TITLE
Rebirth integration tests 

### PR DESCRIPTION
**Pull request by meganokeefe**
- [x] couchbase- @adamschaub i fixed the test i'd had commented out before, it was a meta/ID matching ish.
- [x] docker 
- [x] nomad (see todo)
- [x] auth - both tests were br0ke

(manager functional tests will merge separately after rebirth) 

### TODO 
the nomad end-to-end integration test, `TestNomad` is nondeterministic. have to figure out why sometimes (About 10% of the time) jobs go straight from Inactive->Finished without ever hitting the "Running" state. Seems to only be occurring on travis, not locally. will address in a separate PR

### CODE REVIEW

File: api/v1/server/server.go [hunk 1]
```diff
@@ -558,7 +558,6 @@ func (s *Server) AddServiceOperationProgress(w http.ResponseWriter, r *http.Requ
 		http.Error(w, "no service-operation-id in AddServiceOperationProgress request", http.StatusBadRequest)
 		return
 	}
-
```
Comment (adamschaub) [inline api/v1/server/server.go]: COMPLETELY

File: api/v1/server/server.go [hunk 2]
```diff
@@ -572,6 +571,7 @@ func (s *Server) AddServiceOperationProgress(w http.ResponseWriter, r *http.Requ
 		http.Error(w, "No progress found in AddServiceOperationProgress  request", http.StatusBadRequest)
 		return
 	}
+
```
Comment (adamschaub) [inline api/v1/server/server.go]: INCONSISTENT

File: database/couchbase.go [hunk 3]
```diff
@@ -521,7 +521,6 @@ func (cb *Couchbase) QueryServiceOperationsIds(s model.ServiceOperation, filters
 	if err != nil {
 		return nil, err
 	}
-
```
Comment (adamschaub) [inline database/couchbase.go]: Undo. Avoid just white space additions

File: task/docker.go [hunk 4]
```diff
@@ -147,17 +147,15 @@ func (d *DockerTasker) Run(so *model.ServiceOperation, sites []model.Site) error
 	if so == nil || len(so.Service.GetID()) == 0 {
 		return ErrNilServiceOperation
 	}
-	var env []string
 	cmd := []string{}
 	if so.Operation != "" {
 		cmd = append(cmd, strings.ToLower(so.Operation))
 	}
 
-	cfg2 := &container.Config{
```
Comment (adamschaub) [inline task/docker.go]: what donut named this cfg2

File: task/nomad.go [hunk 5]
```diff
@@ -265,8 +266,7 @@ func artifactsFromSecretMap(secretDir string, svcVars pluginClient.ServiceVars)
 	if len(secretDir) == 0 || svcVars == nil {
 		return nil
 	}
-
-	artifacts := make([]*nomadapi.TaskArtifact, len(svcVars))
+	artifacts := []*nomadapi.TaskArtifact{}
```
Comment (adamschaub) [inline task/nomad.go]: Let's keep make but make len 0 and cap `len(svcVars)`

File: task/docker.go (hunk 4 above)
Comment (mikeynap) [inline task/docker.go]: *raises hand* probably.

**Pull request merged**